File: vipclaw-admin/src/main/java/com/vipclaw/admin/service/ModelProviderService.java
package com.vipclaw.admin.service;

import com.baomidou.mybatisplus.extension.plugins.pagination.Page;
import com.baomidou.mybatisplus.extension.service.IService;
import com.vipclaw.admin.dto.ModelProviderCreateRequest;
import com.vipclaw.admin.dto.ModelProviderResponse;
import com.vipclaw.admin.dto.ModelProviderUpdateRequest;
import com.vipclaw.admin.entity.ModelProvider;

/**
 * 模型服务商服务接口
 *
 * @author vipamp
 * @since 2026-03-13
 */
public interface ModelProviderService extends IService<ModelProvider> {

    /**
     * 分页查询模型服务商
     *
     * @param page     分页对象
     * @param name     服务商名称
     * @param status   状态
     * @return 分页结果
     */
    Page<ModelProviderResponse> page(Page<ModelProvider> page, String name, Integer status);

    /**
     * 获取模型服务商详情
     *
     * @param id ID
     * @return 模型服务商响应
     */
    ModelProviderResponse getDetail(Long id);

    /**
     * 创建模型服务商
     *
     * @param request 创建请求
     * @return 模型服务商响应
     */
    ModelProviderResponse create(ModelProviderCreateRequest request);

    /**
     * 更新模型服务商
     *
     * @param id      ID
     * @param request 更新请求
     * @return 模型服务商响应
     */
    ModelProviderResponse update(Long id, ModelProviderUpdateRequest request);

    /**
     * 切换模型服务商状态
     *
     * @param id ID
     * @return 模型服务商响应
     */
    ModelProviderResponse toggle(Long id);

    /**
     * 连接测试
     *
     * @param id ID
     * @return 是否连接成功
     */
    boolean connectivityTest(Long id);

    /**
     * 删除模型服务商（带校验）
     *
     * @param id ID
     * @return 是否删除成功
     */
    boolean removeProviderById(Long id);
}

File: harnax-admin/src/main/kotlin/com/agnetix/harnax/admin/service/ModelProviderService.kt
package com.agnetix.harnax.admin.service

import com.agnetix.harnax.admin.dto.ModelProviderCreateRequest
import com.agnetix.harnax.admin.dto.ModelProviderResponse
import com.agnetix.harnax.admin.dto.ModelProviderUpdateRequest
import com.agnetix.harnax.admin.dto.ModelStatsInfo
import com.agnetix.harnax.admin.dto.Page
import com.agnetix.harnax.entity.ModelProvider

/**
 * Model provider service interface
 */
interface ModelProviderService {

    /**
     * Query model providers with pagination
     *
     * @param name     Provider name
     * @param type     Provider type
     * @param status   Status
     * @param isPublic Is public
     * @param pageNum  Current page number
     * @param pageSize Page size
     * @return Paginated result
     */
    fun page(name: String?, type: String?, status: Int?, isPublic: Int?, pageNum: Int, pageSize: Int): Page<ModelProvider>

    /**
     * Get model provider details
     *
     * @param id ID
     * @return Model provider response
     */
    fun getModelProvider(id: Long): ModelProvider?

    /**
     * Create model provider
     *
     * @param request Create request
     * @return Model provider response
     */
    fun createModelProvider(request: ModelProviderCreateRequest): Boolean

    /**
     * Update model provider
     *
     * @param id      ID
     * @param request Update request
     * @return Update result
     */
    fun updateModelProvider(id: Long, request: ModelProviderUpdateRequest): Boolean

    /**
     * Update model provider status
     *
     * @param id     ID
     * @param status Status (0:disabled, 1:enabled)
     * @return Update result
     */
    fun updateStatus(id: Long, status: Int): Boolean

    /**
     * Toggle model provider status
     *
     * @param id     ID
     * @param status Status (0:disabled, 1:enabled)
     * @return Update result
     */
    fun toggleModelProvider(id: Long, status: Int): Boolean

    /**
     * Connection test
     *
     * @param id ID
     * @return Whether connection is successful
     */
    fun connectivityTest(id: Long): Boolean

    /**
     * Delete model provider (with validation)
     *
     * @param id ID
     * @return Whether delete is successful
     */
    fun deleteModelProvider(id: Long): Boolean

    /**
     * Convert model provider to response object
     *
     * @param modelProvider Model provider
     * @return Model provider response
     */
    fun convertToResponse(modelProvider: ModelProvider): ModelProviderResponse

    /**
     * Get model statistics for provider
     *
     * @param providerId Provider ID
     * @return Model statistics info
     */
    fun getModelStats(providerId: Long): ModelStatsInfo
}
